File: srcs/server/services/onboarding/dayone.go
package onboarding

import (
	"context"
	"fmt"
)

// RunDayOneSetup orchestrates the Day One setup process for OHC.
func RunDayOneSetup(ctx context.Context, isCloud bool) (string, error) {
	// 1. Provision environment
	if err := ProvisionEnvironment(ctx, isCloud); err != nil {
		return "", fmt.Errorf("provisioning failed: %w", err)
	}

	// 2. Interactive setup
	wizard := NewInteractiveWizard()
	config, err := wizard.RunInteractiveSetup(ctx, isCloud)
	if err != nil {
		return "", fmt.Errorf("interactive setup failed: %w", err)
	}

	// 3. Validate config
	validator := &ValidationEndpoint{}
	if err := validator.ValidateConfig(ctx, config); err != nil {
		return "", fmt.Errorf("config validation failed: %w", err)
	}

	// 4. Generate audit report
	report := GenerateAuditReport(isCloud)
	return report, nil
}

File: src/server/src/services/onboarding/dayone.rs
use crate::services::onboarding::provisioner;
use crate::services::onboarding::wizard::InteractiveWizard;
use crate::services::onboarding::validation::ValidationEndpoint;
use crate::services::onboarding::audit;

pub fn run_day_one_setup(is_cloud: bool) -> Result<String, String> {
    // 1. Provision environment
    provisioner::provision_environment(is_cloud)?;

    // 2. Interactive setup
    let wizard = InteractiveWizard::new();
    let config = wizard.run_interactive_setup(is_cloud)?;

    // 3. Validate config
    let validator = ValidationEndpoint;
    validator.validate_config(&config)?;

    // 4. Generate audit report
    let report = audit::generate_audit_report(is_cloud);

    Ok(report)
}

#[cfg(test)]
mod tests {
    use super::*;
    use std::fs;

    #[test]
    fn test_run_day_one_setup_standalone() {
        let _ = fs::remove_dir_all(".ohc-local-data");

        let report = run_day_one_setup(false).unwrap();
        assert!(report.contains("PASSED"));
        assert!(report.contains("Standalone"));

        fs::remove_dir_all(".ohc-local-data").unwrap();
    }

    #[test]
    fn test_run_day_one_setup_cloud() {
        let _ = fs::remove_dir_all(".ohc-cloud-data");

        let num_cpus = std::thread::available_parallelism().map(|n| n.get()).unwrap_or(1);

        let res = run_day_one_setup(true);
        if num_cpus < 2 {
            assert!(res.is_err());
            assert!(res.unwrap_err().contains("preflight check failed"));
        } else {
            let report = res.unwrap();
            assert!(report.contains("PASSED"));
            assert!(report.contains("Cloud-native"));
        }

        let _ = fs::remove_dir_all(".ohc-cloud-data");
    }
}
